Sources/Markdown/Base/Markup.swift

```diff
@@ -186,11 +186,22 @@ extension Markup {
     /// - Complexity: `O(childCount)`
     public func child(at position: Int) -> Markup? {
         precondition(position >= 0, "Cannot retrieve a child at negative index: \(position)")
-        guard position <= raw.markup.childCount else {
+        guard position < raw.markup.childCount else {
             return nil
         }
-        var iterator = children.dropFirst(position).makeIterator()
-        return iterator.next()
+        
+        let childMetadata: MarkupMetadata
+        if position == 0 {
+            childMetadata = raw.metadata.firstChild()
+        } else {
+            let siblings = (0..<position).map(raw.markup.child(at:))
```

Comment:
`RawMarkup` has a tail-allocated array `.children` which you can use here to quickly get to the next siblings. Try not to make an additional array here if you can: `raw.markup.children.prefix(position).reduce` etc.

Reply:
It looks like `children` is just a computed property that calls `child(at:)`?

```swift
var children: AnySequence<RawMarkup> {
    return AnySequence((0..<childCount).lazy.map { Int -> RawMarkup in
        self.child(at: 0)
    })
}
```

Wouldn't it be better to specifically request the small `0..<position` array instead of `0..<children` since this will result in a new array either way?

Reply:
This logic changed slightly when I addressed your other feedback- let me know what you think.

Reply:
What I mean is, if all you are doing is accumulating the child counts, the tail allocated children of the raw markup type is there for fast, direct access. So if you only need to loop, try to do it directly.

Reply:
Sorry- I'm still not totally following. The `child(at:)` method appears to do direct access to me. How would you modify the current logic?

Reply:
Looking at your latest change, you’ve got it.